src/Mappers/InventoryMapper.php

```diff
@@ -110,19 +111,48 @@ public function createInventoryDTOsFromVariation($variationData)
         continue;
       }
 
+      // Avl Immediately. Net Stock.
+      $onHand = isset($stock['netStock']) ? $stock['netStock'] : null;
+      // Already Ordered items.
+      $onOrder = isset($stock['reservedStock']) ? $stock['reservedStock'] : null;
+
+      // key for a potential preexisting DTO that we need to merge with
+      $dtoKey = $supplierId . '_' . $supplierPartNumber;
+
+      /** @var RequestDTO $existingDTO */
+      $existingDTO = $requestDtosBySuID[$dtoKey];
+
+      if (isset($existingDTO) && !empty($existingDTO))
+      {
+        /* merge with previous values for this suID.
+         * stock values should be summed.
+         * Variation-level data (nextAvailableDate) does not vary.
+        */
+        if (null !== $existingDTO->getQuantityOnHand())
+        {
+          $onHand += $existingDTO->getQuantityOnHand();
+        }
```

Comment:
As -1 can be an option we will need to handle this scenario. My initial thoughts are:

- If both are -1 then quantity is -1
- If one is -1 and the other is 0 then quantity is -1
- If one is -1 and the other is greater than 0 then quantity is the positive quantity

Reply:
@mcloughlin77 check out the new static function and associated test